src/script/components/utils/InViewport.tsx

```diff
@@ -25,6 +25,7 @@ import {viewportObserver} from 'Util/DOM/viewportObserver';
 interface InViewportParams {
   onVisible: () => void;
```

Comment:
in the current implementation, this callback will be called when most of the element is visible. and we need it this way to mark the message as read. if I reuse this callback, the button "jump to the last message" will be disappearing slightly different from how it behaves on our android app - not when a bit of the last message appears, but when the big chunk of it is visible. @atomrc suggested we leave it as it is. the other option would be to implement one callback `onPartialVisibilityChange(visible: boolean)` instead of `onVisibilityLost()` and call it with `true` differently than `onVisible` (as `onVisible` is used in many places, not only for messages)